### PR TITLE
Fix logic for which docs sections are open

### PR DESCRIPTION
**Pull request by wackerow**
## Description
- Actively tracks state of which accordion items should be open or close
- Allows us to force reset this when the page path changes, defaulting back to only the section containing the active article being open. 

## Preview link demo
1. https://deploy-preview-194--melodious-puffpuff-8e1109.netlify.app/docs/fundamentals/account-management
2. Click on **Clef** link in first paragraph of page content
3. See "Fundamentals" section close and "Tools" section open upon navigation

## Related issue
- Patches the fix for #164

### CODE REVIEW

**Comment by wackerow** [comment]
https://user-images.githubusercontent.com/54227730/208532531-c0a5e9a6-3e88-47a9-a697-c8c4a8ccf0d2.mov

**Review by corwintines — APPROVED** [review]
Nice

File: src/components/UI/docs/DocsLinks.tsx [hunk 1]
```diff
@@ -24,16 +24,37 @@ interface Props {
...
+    setOpenSections(prev => ({ ...prev, [id]: !prev[id] }));
+  };
+
   return (
     <Stack border='2px' borderColor='primary'>
       {navLinks.map(({ id, to, items }, idx) => {
         const split = to?.split('/');
         const isActive = slug && split && split[split.length - 1] === slug[slug.length - 1];
-        const isSectionActive = checkNavLinks({ to, items, pathCheck: asPath.split('#')[0] })
+        const isSectionActive = openSections[id];
```
Comment (wackerow) [inline src/components/UI/docs/DocsLinks.tsx:54]: If we're going to bother extracting this, let's just extract all of the logic
```suggestion
        const index = openSections[id] ? 0 : -1;
```

File: src/components/UI/docs/DocsLinks.tsx [hunk 2]
```diff
@@ -24,16 +24,37 @@ interface Props {
...
 
         return (
-          <Accordion key={id} defaultIndex={isSectionActive ? 0 : -1} allowToggle mt='0 !important'>
+          <Accordion key={id} index={isSectionActive ? 0 : -1} allowToggle mt='0 !important'>
```
Comment (wackerow) [inline src/components/UI/docs/DocsLinks.tsx:57]: With above
```suggestion
          <Accordion key={id} index={index} allowToggle mt='0 !important'>
```

File: src/components/UI/docs/DocsLinks.tsx (hunk 1 above)
Comment (wackerow) [reply src/components/UI/docs/DocsLinks.tsx:54]: Happy to ignore too and just get this in =)

**Pull request merged**